config/plugins/pbh.py

```diff
@@ -0,0 +1,816 @@
+    """
+
+    if value is not None:
+        try:
+            ip = ipaddress.ip_address(value)
+        except Exception as e:
+            exit_with_error("Error: invalid value '{}' for '{}' option\n{}".format(value, param.name, e), fg="red")
+
+        return str(ip)
+
+
+def re_match(value, param_name, regexp):
+    """ Regexp validation of given parameter
+
+        Args:
+            value: value to validate,
+            param_name: parameter name,
+            regexp: regular expression
+
+        Return:
+            str: validated value
+    """
+
+    if re.match(regexp, str(value)) is None:
+        exit_with_error("Error: invalid value '{}' for '{}' option".format(str(value), param_name), fg="red")
+
+    return value
+
+
+def pbh_re_match(ctx, param, value):
+    """ Check if PBH rule options are valid
+
+        Args:
+            ctx: click context,
+            param: click parameter context,
+            value: value of parameter
+
+        Returns:
+            str: validated parameter
+    """
+
+    if value is not None:
+        if param.name == 'gre_key':
+            return re_match(value, param.name, gre_key_re)
+        elif param.name == 'ip_protocol':
+            return re_match(value, param.name, ip_protocol_re)
+        elif param.name == 'ipv6_next_header':
+            return re_match(value, param.name, ipv6_next_header_re)
+        elif param.name == 'l4_dst_port':
+            return re_match(value, param.name, l4_dst_port_re)
+        elif param.name == 'inner_ether_type':
+            return re_match(value, param.name, inner_ether_type_re)
+
+
+def is_exist_in_db(db, obj_list, conf_db_key):
+    """ Check if provided CLI option already exist in Config DB,
+        i.g in case of --hash-field-list option it will check
+        if 'hash-field' was previously added by
+        'config pbh hash-field ...' CLI command
+
+        Args:
+            db: reference to Config DB,
+            obj_list: value of 'click' option
+            conf_db_key: key to search in Config DB
+    """
+
+    if obj_list is None:
+        return True
+
+    table = db.cfgdb.get_table(conf_db_key)
+    correct_list = list(table.keys())
+
+    splited_list = obj_list.split(',')
+
+    for elem in splited_list:
+        if elem not in correct_list:
+            return False
+
+    return True
+
+
+def ip_mask_hash_field_correspondence(ip_mask, hash_field):
+    """ Check if the --ip-mask option are correspond to
+        the --hash-field option
+
+        Args:
+            ip_mask: ip address,
+            hash_field: hash field value, which was configured before
+    """
+
+    is_error = False
+    hf_v4_and_v6 = ['INNER_DST_IPV4', 'INNER_SRC_IPV4', 'INNER_DST_IPV6', 'INNER_SRC_IPV6']
+    hf_no_ip = ['INNER_IP_PROTOCOL', 'INNER_L4_DST_PORT', 'INNER_L4_SRC_PORT']
+    hf_v4 = ['INNER_DST_IPV4', 'INNER_SRC_IPV4']
+    hf_v6 = ['INNER_DST_IPV6', 'INNER_SRC_IPV6']
+
+    if (ip_mask is None) and (hash_field in hf_v4_and_v6):
```

Comment:
@vadymhlushko-mlnx what if `hash_field` is `None`? Do we guarantee that only `ip_mask` can be `None` inside this method?